File: freecodecamp/daily-code-challenge/2026-04-11.py
def rook_bishop_attack(rook, bishop):
    # Convert columns (A-H) to numbers (1-8)
    col1, row1 = ord(rook[0]) - ord('A'), int(rook[1])
    col2, row2 = ord(bishop[0]) - ord('A'), int(bishop[1])
    
    # Rook attack (same row or column)
    if col1 == col2 or row1 == row2:
        return "rook"
    
    # Bishop attack (diagonal)
    if abs(col1 - col2) == abs(row1 - row2):
        return "bishop"
    
    return "neither"

File: freecodecamp/daily-code-challenge/2026-04-11.js
function rookBishopAttack(rook, bishop) {
  // Convert column letters to numbers (A → 0, B → 1, ..., H → 7)
  const col1 = rook.charCodeAt(0) - 'A'.charCodeAt(0);
  const row1 = parseInt(rook[1]);

  const col2 = bishop.charCodeAt(0) - 'A'.charCodeAt(0);
  const row2 = parseInt(bishop[1]);

  // Rook attack (same row or column)
  if (col1 === col2 || row1 === row2) {
    return "rook";
  }

  // Bishop attack (diagonal)
  if (Math.abs(col1 - col2) === Math.abs(row1 - row2)) {
    return "bishop";
  }

  return "neither";
}